### ISSUE TITLE
Debugger: Func-eval of function returning a 'decimal' type is broken on macOS/Linux

### ISSUE THREAD

**Issue by gregg-miskelly** [dotnet/coreclr#6935]
#### Steps to Reproduce:
Create a .NET Core Console application with the following code:

```
namespace ConsoleApplication
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var t = new Test { Value = 1.0m };
            // SET BREAKPOINT HERE
            System.Console.WriteLine(t.Value);
        }
    }

    public class Test
    {
        public decimal Value
        {
            get { return _value; }
            set { _value = value;}
        }

        private decimal _value;
    }
}

Stop on the set breakpoint here line
Attempt to func-eval t.Value

#### Result:
On Linux/macOS: If the debugger takes the ICorDebugValue returned from the evaluation, unboxes it, and reads the memory which the resulting value, the lower 64-bits are fine, but the high 64-bit look corrupt.
On Windows: Everything works

**Comment by gregg-miskelly** [comment]
@noahfalk @Mikem8361 here is the new bug

**Comment by gregg-miskelly** [comment]
We have had several reports of this, so it would be nice to fix for 1.1 if that is possible.

**Comment by mikem8361** [comment]
Fixed with PR #6997.
